### ISSUE TITLE
Improved Autocomplete Hinter

### ISSUE THREAD

**Issue by RandomGamingDev** [processing/p5.js-web-editor#3078]
### Increasing Access

This would make it easier to code in the p5.js editor, lessen the irritation of typos for beginners, as well as improving the quality of life for developers in general. This is because the current Autocomplete Hinter doesn't hint at variables, functions, or classes created by the user, and mostly just serves as (and this is a personal belief) a worse version of the p5.js documentation that clogs up the screen by covering code, and that doesn't show examples or explain how a function works.

### Feature enhancement details

- Add autocomplete hints for variables, functions, and classes
- Add an option to access the documentation for interacting with code belonging to p5.js
- Add an option to jump to the variable, function, or class when interacting with code belonging to the sketch itself
- Maybe try something like ace.js's approach (here are some links talking about ace.js's autocomplete)
a. https://github.com/ajaxorg/ace/blob/e59ae67d1c348ea6cb6b2b112b4a504a09fabfba/src/autocomplete.js#L68
b. https://stackoverflow.com/questions/13545433/autocompletion-in-ace-editor
- Here's the previous issue mentioning adding the Autocomplete Hinter and which resulted in the addition of the current hinter: https://github.com/processing/p5.js-web-editor/issues/99

**Comment by lindapaiste** [comment]
> * Add an option to access the documentation for interacting with code belonging to p5.js

This should be the easiest one to implement, so probably a good starting point.

**Comment by suprabhat15** [comment]
> * Add autocomplete hints for variables, functions, and classes
> * Add an option to access the documentation for interacting with code belonging to p5.js

Hints are available and Clicking on arrow redirects users to the docs of p5.js. So, as per the following image, these 2 features are implemented, right?

@RandomGamingDev @lindapaiste Please let me know if I am missing something here.

<img width="298" alt="image" src="https://github.com/user-attachments/assets/b968e147-5f41-4c8d-a92b-15c695a50dfb">

**Comment by RandomGamingDev** [comment]
> > * Add autocomplete hints for variables, functions, and classes
> > * Add an option to access the documentation for interacting with code belonging to p5.js
> 
> Hints are available and Clicking on arrow redirects users to the docs of p5.js. So, as per the following image, these 2 features are implemented, right?
> 
> @RandomGamingDev @lindapaiste Please let me know if I am missing something here.
> <img alt="image" width="298" src="https://private-user-images.githubusercontent.com/47072868/363438873-b968e147-5f41-4c8d-a92b-15c695a50dfb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxMzY2NTMsIm5iZiI6MTcyNTEzNjM1MywicGF0aCI6Ii80NzA3Mjg2OC8zNjM0Mzg4NzMtYjk2OGUxNDctNWY0MS00YzhkLWE5MmItMTVjNjk1YTUwZGZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMxVDIwMzIzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlMDhkODAzMjQ1MTNiNjBiN2Q1YzI2YWFjZDI3ZTUxYmMxZTRhMmJiMzUyM2ZkYTJjMjMwNjg2MWM0MmYxNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6IyNT5GLZzT0Hh1UMgqjA77i5ct64bsL96ScGoAVB2c">

No, because things like hinting for variables and functions created by the user or from other libraries aren't implemented yet

**Comment by 29deepanshutyagi** [comment]
i want to work on this issue if it's still open @andrewn
